flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/allocator/SlotSharingSlotAllocator.java

```diff
@@ -133,7 +138,56 @@ public Optional<VertexParallelismWithSlotSharing> determineParallelism(
         return Optional.of(new VertexParallelismWithSlotSharing(allVertexParallelism, assignments));
     }
 
-    private static Map<JobVertexID, Integer> determineParallelism(
+    /**
+     * Distributes free slots across the slot-sharing groups of the job. Slots are distributed as
+     * evenly as possible while taking the minimum parallelism of contained vertices into account.
```

Comment:
```suggestion
     * evenly as possible while taking the lowest possible parallelism of contained vertices into account (maximum out of explicitly configured parallelisms).
```

Reply:
> (maximum out of explicitly configured parallelisms)

will rephrase the related sentence because it will become difficult to wrap your head around it.